src/components/lineup/Lineup.tsx

```diff
@@ -0,0 +1,295 @@
+        pauseTrack()
+        track(
+          make({
+            eventName: Name.PLAYBACK_PAUSE,
+            id: `${id}`,
+            source: source || PlaybackSource.TRACK_TILE
+          })
+        )
+      }
+    },
+    [playTrack, pauseTrack, playing, playingUid]
+  )
+
+  const renderItem = ({ index, item }: { index: number; item: LineupItem }) => {
+    if (item._loading) {
+      return (
+        <View style={styles.item}>
+          <TrackTileSkeleton />
+        </View>
+      )
+    }
+
+    if (item.kind === Kind.TRACKS || item.track_id) {
+      // Render a track tile if the kind tracks or there's a track id present
+
+      if (item._marked_deleted) {
+        return null
+      }
+
+      return (
+        <View style={styles.item}>
+          <TrackTile
+            {...item}
+            index={index}
+            isTrending={isTrending}
+            showArtistPick={showLeadingElementArtistPick && !!leadingElementId}
+            showRankIcon={index < rankIconCount}
+            togglePlay={togglePlay}
+            uid={item.uid}
+          />
+        </View>
+      )
+    } else {
+      return null
+      // TODO: Playlist tile
+    }
+  }
+
+  // Calculate the sections of data to provide to SectionList
+  const sections = useMemo(() => {
+    const { entries, hasMore, isMetadataLoading, page } = lineup
+    const itemDisplayCount = page <= 1 ? itemCounts.initial : pageItemCount
+
+    const getSkeletonCount = () => {
+      if (
+        isMetadataLoading &&
+        hasMore &&
+        entries.length < countOrDefault &&
+        (!limit || entries.length !== limit)
+      ) {
+        // Calculate the number of skeletons to display: total # requested - # rendered - # deleted
+        // If the `count` prop is provided, render the count - # loaded tiles
+        const loadingSkeletonDifferential = Math.max(
+          itemDisplayCount - entries.length - lineup.deleted,
+          0
+        )
+        return count
+          ? Math.min(count - entries.length, MAX_COUNT_LOADING_TILES)
+          : loadingSkeletonDifferential
+      }
+      return 0
+    }
+
+    // Append loading items to the array of already loaded items
+    const items = [
+      ...entries,
+      ...range(getSkeletonCount()).map(() => ({ _loading: true } as LineupItem))
+    ]
+
+    // If delineate=true, create sections of items based on time.
+    // Otherwise return one section
+    return delineate
+      ? delineateByTime(items)
+      : [
+          {
+            title: '',
+            data: items
+          }
+        ]
+  }, [
+    count,
+    countOrDefault,
+    delineate,
+    itemCounts,
+    limit,
+    lineup,
+    pageItemCount
+  ])
+
+  return (
+    <SectionList
+      ListHeaderComponent={header}
+      ListFooterComponent={<View style={{ height: 160 }} />}
+      onEndReached={onEndReached}
+      onEndReachedThreshold={LOAD_MORE_THRESHOLD}
+      // TODO: Either style the refreshing indicator or
```

Comment:
This will be fun. Our existing pull to refresh never was perfect on the web view, but I think RN is a great opportunity to make it super slick. Potentially @dylanjeffers could tackle with the profile page pull-to-refresh since that has some nice interplay with the cover photo expanding / fading